Activities/Cryptography/UiPath.Cryptography.Activities/KeyedHashFile.cs

```diff
@@ -92,34 +99,46 @@ protected override string Execute(CodeActivityContext context)
 
             try
             {
-                string filePath = FilePath.Get(context);
-                string key = Key.Get(context);
-                SecureString keySecureString = KeySecureString.Get(context);
-                Encoding keyEncoding = Encoding.Get(context);
+                var filePath = FilePath.Get(context);
+                var key = Key.Get(context);
+                var keySecureString = KeySecureString.Get(context);
+                var keyEncoding = Encoding.Get(context);
+                var inputFile = InputFile.Get(context);
 
-                if (string.IsNullOrWhiteSpace(filePath))
-                {
+                if (string.IsNullOrWhiteSpace(filePath) && inputFile == null)
                     throw new ArgumentNullException(Resources.FilePathDisplayName);
-                }
+
                 if (string.IsNullOrWhiteSpace(key) && keySecureString == null)
-                {
                     throw new ArgumentNullException(Resources.KeyAndSecureStringNull);
-                }
+
+                //either input file path or input file as resource should be used
+                if (!string.IsNullOrWhiteSpace(filePath) && inputFile != null)
+                    throw new ArgumentNullException(string.Format(Resources.Exception_UseOnlyFilePathOrInputResource,
+                        Resources.Activity_KeyedHashFile_Property_InputFile_Name, Resources.Activity_KeyedHashFile_Property_FilePath_Name));
+
                 if (key != null && keySecureString != null)
-                {
                     throw new ArgumentNullException(Resources.KeyAndSecureStringNotNull);
-                }
+
                 if (keyEncoding == null)
-                {
                     throw new ArgumentNullException(Resources.Encoding);
-                }
-                if (!File.Exists(filePath))
-                {
+
+                if (!File.Exists(filePath) && inputFile == null)
                     throw new ArgumentException(Resources.FileDoesNotExistsException, Resources.FilePathDisplayName);
-                }
 
+                byte[] hashed;
+
+                if (inputFile != null && !inputFile.IsFolder)
```

Comment:
Shouldn't we throw an exception or something if the input file is a folder?

Same for other activities as well.

Reply:
done